### ISSUE TITLE
Rule MD024 should be refined [to allow heading duplication for non-siblings]

### ISSUE THREAD

**Issue by bfzhao** [DavidAnson/markdownlint#136]
It is reasonable to have rule MD024 to report any title conflict, but it makes sense only when those titles have the same parent title. Below case is very common and should not be reported as an issue:
```
# cast 1
## RCA
## RCA -----> should report here
## Action List 
# case 2
## RCA -----> should not report here
## Action List 
```

**Comment by DavidAnson** [comment]
Although this conflicts with the rationale given in the rule description (https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md024---multiple-headings-with-the-same-content), it seems reasonable as an option.
